apps/asap-server/test/handlers/webhooks/webhook-events-updated.test.ts

```diff
@@ -1,16 +1,204 @@
+import { RestCalendar } from '@asap-hub/squidex';
+import { SyncCalendarFactory } from '../../../src/utils/sync-google-calendar';
+
+const googleCalendarId = 'calendar-id@group.calendar.google.com';
+const squidexCalendarId = 'squidex-calendar-id';
+
+const updateCalendarResponse: CalendarResponse = {
+  id: googleCalendarId,
+  color: '#5C1158' as GoogleLegacyCalendarColor,
+  name: 'Kubernetes Meetups',
+};
+
+const fetchCalendarRawResponse: RestCalendar = {
+  id: squidexCalendarId,
+  data: {
+    id: { iv: googleCalendarId },
+    color: { iv: '#5C1158' as GoogleLegacyCalendarColor },
+    name: { iv: 'Kubernetes Meetups' },
+    syncToken: { iv: 'google-sync-token' },
+  },
+  created: '2021-01-07T16:44:09Z',
+  lastModified: '2021-01-07T16:44:09Z',
+};
 
 describe('Event Webhook', () => {
-  test('Should not fail', async () => {
-    expect(
-      handler(
-        apiGatewayEvent({
-          body: undefined,
-        }),
-        {} as any,
-        undefined as any,
-      ),
-    ).resolves.not.toThrow();
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  const getJWTCredentialsMock: jest.MockedFunction<GetJWTCredentials> = jest
+    .fn()
+    .mockResolvedValue({
+      client_email: 'random-data',
+      private_key: 'random-data',
+    });
+
+  const syncCalendarMock = jest.fn();
+  const syncCalendarFactoryMock: jest.MockedFunction<SyncCalendarFactory> = jest
+    .fn()
+    .mockReturnValue(syncCalendarMock);
+
+  const handler = webhookEventUpdatedHandlerFactory(
+    calendarControllerMock,
+    getJWTCredentialsMock,
+    syncCalendarFactoryMock,
+  );
+
+  calendarControllerMock.getSyncToken.mockResolvedValue('next-sync-token');
+
+  test('Should return 400 when x-goog-resource-id is not set', async () => {
+    const res = (await handler(
+      apiGatewayEvent({
+        ...googlePayload,
+        headers: {},
+      }),
+    )) as APIGatewayProxyResult;
+
+    expect(res.statusCode).toStrictEqual(400);
+  });
+
+  test('Should return 502 when fails to get calendar from squidex', async () => {
+    calendarControllerMock.fetchByResouceId.mockRejectedValueOnce(
+      new Error('Squidex Error'),
+    );
+
+    const res = (await handler(
+      apiGatewayEvent(googlePayload),
+    )) as APIGatewayProxyResult;
+
+    expect(res.statusCode).toStrictEqual(502);
+  });
+
+  test('Should return 502 when fails to get credentials from AWS', async () => {
+    calendarControllerMock.fetchByResouceId.mockResolvedValueOnce(
+      fetchCalendarRawResponse,
+    );
+    calendarControllerMock.getSyncToken.mockResolvedValueOnce(undefined);
+    getJWTCredentialsMock.mockRejectedValueOnce(new Error('AWS Error'));
+
+    const res = (await handler(
+      apiGatewayEvent(googlePayload),
+    )) as APIGatewayProxyResult;
+
+    expect(res.statusCode).toStrictEqual(502);
+  });
+
+  test('Should return 200 and do full synk when it fails to fetch the synkToken from squidex', async () => {
```

Comment:
sync?

Reply:
It's like sync, just a bit more radikal 🤘🏽